### PR TITLE
CRM-17350 - first pass at allowing editing of tags when you don't hav…

### PR DESCRIPTION
**Pull request by MegaphoneJon**
…e 'edit all contacts' permission

### CODE REVIEW

File: CRM/Core/DAO/permissions.php [hunk 1]
```diff
@@ -114,7 +114,6 @@ function _civicrm_api3_permissions($entity, $action, &$params) {
     ),
     'default' => array(
       'access CiviCRM',
-      'edit all contacts',
```
Comment (seamuslee001) [inline CRM/Core/DAO/permissions.php]: Won't this affect a bunch of other entities apart from entity_tag?

Comment (PalanteJon) [inline CRM/Core/DAO/permissions.php]: oh jeez.  That was meant to be temporary!  Yes, that's a dumb move.  I'll correct it.

**Comment by colemanw** [comment]
This seems like it will work well when the entity is "contact" but would need a fallback when tagging other things.

**Comment by PalanteJon** [comment]
That makes sense.  I'll work on that later in the week and add to this PR.

**Comment by colemanw** [comment]
Thanks @PalanteJon ping me with any questions.

**Comment by eileenmcnaughton** [comment]
I have pulled your fix into this (tested) api https://github.com/civicrm/civicrm-core/pull/7649 - but it turns out the api does not support creating entities other than contact. (although it just occurred to me I need to check that check_permissions is on with a test too)

**Pull request closed without merging**